tools/aa/commands/SyncCommand.java

```diff
@@ -48,11 +57,49 @@ public void run(String[] args) {
           .filter(path -> fileUtils.folderExists(path))
           .forEach(
               path -> {
-                System.out.println("Performing sync: " + path.toString());
+                System.out.println(String.format("[HEAD]: Performing sync: %s", path));
                 repoFactory.create(path).pull();
               });
     } catch (IOException e) {
       e.printStackTrace();
     }
+    // then, do the sync for all workspaces
+    String workspacePath = fileUtils.joinPaths(config.getBasePath(), "ws", currentWorkspaceName);
+    try {
+      fileUtils
+          .listContents(workspacePath)
+          .stream()
+          .map(path -> fileUtils.joinPaths(workspacePath, path))
+          .filter(path -> fileUtils.folderExists(path))
+          .forEach(
+              path -> {
+                String repoName = Paths.get(path).getFileName().toString();
+                System.out.println(
+                    String.format("[%s/%s]: Performing sync", currentWorkspaceName, repoName));
+                GitRepo repo = repoFactory.create(path);
+                System.out.println(
+                    String.format(
+                        "[%s/%s]: switching to temp branch", currentWorkspaceName, repoName));
+                repo.switchBranch("temp_branch_for_sync");
+                System.out.println(
+                    String.format(
+                        "[%s/%s]: committing all changes", currentWorkspaceName, repoName));
+                repo.commit(repo.getUncommittedFiles(), "Sync: temporary commit");
+                System.out.println(
+                    String.format("[%s/%s]: switching to master", currentWorkspaceName, repoName));
+                repo.switchBranch("master");
+                System.out.println(
+                    String.format("[%s/%s]: pulling", currentWorkspaceName, repoName));
+                repo.pull();
+                System.out.println(
+                    String.format("[%s/%s]: merging changes", currentWorkspaceName, repoName));
+                repo.merge("temp_branch_for_sync");
```

Comment:
What happens if the merge fails? Will we get <<<>>> in the file?

Reply:
@oferb this one is a great finding, thanks. With changes in https://github.com/google/startup-os/pull/68/commits/b8bff6d103962b8966ad8541e26f253 we now will